src/App.tsx

```diff
@@ -1,61 +1,28 @@
-import { FC } from 'react';
 import { Switch } from 'react-router-dom';
 
-import { AppProps } from './lib/types/app';
-
-import { useDataContainer } from './lib/hooks/useDataContainer';
+import { useCurrentRoute } from './lib/hooks/useCurrentRoute';
+import { useSubdomainData } from './lib/hooks/useSubdomainData';
 
 import { Domains } from './pages/Domains';
 import { NFT } from './pages/NFT';
 
-import { getDomainId } from './lib/util/domains/domains';
-
 import classNames from 'classnames/bind';
 import styles from './App.module.scss';
 
 const cx = classNames.bind(styles);
 
-export const App: FC<AppProps> = ({ route }) => {
-	const domainId = getDomainId(route);
-	const isRoot = route.split('.').length === 1 && !route.includes('.');
-
-	const {
-		domain,
-		subdomainData,
-		paymentTokenInfo,
-		metrics,
-		domainMetadata,
-		isNFTView,
-		isSubdomainDataLoading,
-	} = useDataContainer(domainId);
+export const App = () => {
+	const { isRootDomain, isNftView, currentDomainId } = useCurrentRoute();
```

Comment:
big fan of this hook !

Reply:
Going to be quite handy - only pain point is that `isNftView` also depends on subdomain data. But I guess that check only needs to happen once so not a big deal!